### PR TITLE
Expose barStyle for NavigatorIOS and TabBarIOS

### PR DESCRIPTION
**Pull request by jacobp100**
Exposes barStyle property. Code already existed in RCTConvert, so that’s why there’s no conversion code here.

### CODE REVIEW

**Comment by ericvicenti** [comment]
Nice! Can you add this to the examples?

**Comment by jacobp100** [comment]
I think we’ve usually just showed a single example with a navigation or tab bar using custom text and background colours. It can’t really be used at the same time as custom background colours, so I can either add separate examples for this, or just post screenshots.

**Comment by ericvicenti** [comment]
I think separate examples make sense if they cannot be demonstrated together. Ideally you could just toggle the `barStyle` dynamically, but I know that isn't always supported.

**Comment by jacobp100** [comment]
screen shot 2016-11-14 at 21 40 26
screen shot 2016-11-14 at 21 40 34

**Comment by mkonicek** [comment]
Looks good, thanks! I expanded the comments a bit.

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
Failed with a Flow error, can you fix it please?

    Libraries/Components/TabBarIOS/TabBarIOS.ios.js line 78 col 30-37: property `barStyle`
    Property not found in
    Libraries/Components/TabBarIOS/TabBarIOS.ios.js line 78 col 19-28: object type

**Comment by jacobp100** [comment]
Just done for TabBarIOS—it seems like there aren’t types for NavigatorIOS.

**Comment by hramos** [comment]
Re-triggering tests. We can't land until this is green.

**Comment by lacker** [comment]
Again re-triggering tests ;-)

**Comment by jacobp100** [comment]
It was failing on flow. I've just remerged master in. Hopefully this time!

**Comment by jacobp100** [comment]
Should be good now!

**Comment by lacker** [comment]
Travis failing on objc-ios, the errors are incomprehensible to me so I am kicking it. https://travis-ci.org/facebook/react-native/builds/199915022

**Comment by jacobp100** [comment]
For the failing bar style, there seems to be an existing comment,

```objc
// TabBarExample on tvOS passes locally but not on Travis
```

I'm not sure why the web socket test is failing. I didn't touch that.

**Comment by jacobp100** [comment]
@lacker We're all good now by the looks of things!

**Comment by jacobp100** [comment]
@lacker Anything I can do my end?

**Comment by hramos** [comment]
As far as I can tell, the import failure is unrelated to the code here. We're having issues importing some of the GitHub PRs this week. I'll import this as soon as they're resolved.

**Comment by ericvicenti** [comment]
This needs to be rebased before it can merge

**Comment by jacobp100** [comment]
Merged in latest master.

**Comment by jacobp100** [comment]
@hramos

**Comment by hramos** [comment]
Ping @ericvicenti who is oncall this week.

**Comment by jacobp100** [comment]
@ericvicenti @hramos Is there anything I can do to help? I can rebase if you want: would you want that on a new branch?

**Comment by jacobp100** [comment]
@hramos @ericvicenti I've updated this to use the new RNTester project, and fixed a flow error. Verified it's still working.

CircleCI seems to be failing on building for Android (looks like some libraries are missing). I've not changed any Android files. Is there anything I need to do?

File: React/Views/RCTTabBar.h [hunk 1]
```diff
@@ -15,5 +15,8 @@
 @property (nonatomic, strong) UIColor *tintColor;
 @property (nonatomic, strong) UIColor *barTintColor;
 @property (nonatomic, assign) BOOL translucent;
+#if !TARGET_OS_TV
+@property (nonatomic, assign) UIBarStyle *barStyle;
```
Comment (shergin) [inline React/Views/RCTTabBar.h]: Please remove `*` here.

File: React/Views/RCTTabBar.m [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 #import "RCTTabBar.h"
 
+#import "RCTConvert.h"
```
Comment (shergin) [inline React/Views/RCTTabBar.m]: Why do we import it `RCTConvert`?

**Comment by jacobp100** [comment]
@shergin Good catches. I've updated the PR.

**Comment by hramos** [comment]
After looking at the internal state of this pull request, it seems like we won't be accepting this change. @shergin feel free to re-open if you think otherwise.

**Comment by jacobp100** [comment]
Is there anything I can do to help this?

**Comment by shergin** [comment]
Okay, let's try one more time!

File: RNTester/js/NavigatorIOSBarStyleExample.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL
+ * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+'use strict';
+
+var React = require('react');
+var ReactNative = require('react-native');
+var {
```
Comment (shergin) [inline RNTester/js/NavigatorIOSBarStyleExample.js]: Could you please fix code style here?

Comment (jacobp100) [inline RNTester/js/NavigatorIOSBarStyleExample.js]: Could you clarify what you wanted changing?

Comment (shergin) [inline RNTester/js/NavigatorIOSBarStyleExample.js]: One module - one line, I think you can import all modules directly by names.

**Pull request closed without merging**